### ISSUE TITLE
loss = nan / 0

### ISSUE THREAD

**Issue by Arthur-Shi** [sfzhang15/RefineDet#98]
I followed the READ.md to convert VOC2007 into lmdb datasets. But I cannot open the mdb files by access and excel, is it right?

And During my training, the losses are nan/0.

Iteration 0, loss = 26.9579
I0726 17:36:25.590222  6123 solver.cpp:259]     Train net output #0: arm_loss = 9.57847 (* 1 = 9.57847 loss)
I0726 17:36:25.590236  6123 solver.cpp:259]     Train net output #1: odm_loss = 17.3795 (* 1 = 17.3795 loss)
I0726 17:36:25.590265  6123 sgd_solver.cpp:138] Iteration 0, lr = 0.001
I0726 17:36:30.275578  6123 solver.cpp:243] Iteration 10, loss = 19.514
I0726 17:36:30.275650  6123 solver.cpp:259]     Train net output #0: arm_loss = 6.04442 (* 1 = 6.04442 loss)
I0726 17:36:30.275662  6123 solver.cpp:259]     Train net output #1: odm_loss = 15.4255 (* 1 = 15.4255 loss)
I0726 17:36:30.275673  6123 sgd_solver.cpp:138] Iteration 10, lr = 0.001
I0726 17:36:35.329936  6123 solver.cpp:243] Iteration 20, loss = nan
I0726 17:36:35.330014  6123 solver.cpp:259]     Train net output #0: arm_loss = nan (* 1 = nan loss)
I0726 17:36:35.330031  6123 solver.cpp:259]     Train net output #1: odm_loss = 0 (* 1 = 0 loss)
I0726 17:36:35.330044  6123 sgd_solver.cpp:138] Iteration 20, lr = 0.001
I0726 17:36:40.332947  6123 solver.cpp:243] Iteration 30, loss = nan
I0726 17:36:40.333024  6123 solver.cpp:259]     Train net output #0: arm_loss = nan (* 1 = nan loss)
I0726 17:36:40.333043  6123 solver.cpp:259]     Train net output #1: odm_loss = 0 (* 1 = 0 loss)
I0726 17:36:40.333058  6123 sgd_solver.cpp:138] Iteration 30, lr = 0.001
I0726 17:36:45.324910  6123 solver.cpp:243] Iteration 40, loss = nan
I0726 17:36:45.324980  6123 solver.cpp:259]     Train net output #0: arm_loss = 0 (* 1 = 0 loss)
I0726 17:36:45.324995  6123 solver.cpp:259]     Train net output #1: odm_loss = 0 (* 1 = 0 loss)
I0726 17:36:45.325007  6123 sgd_solver.cpp:138] Iteration 40, lr = 0.001
I0726 17:36:50.343459  6123 solver.cpp:243] Iteration 50, loss = nan
I0726 17:36:50.343544  6123 solver.cpp:259]     Train net output #0: arm_loss = nan (* 1 = nan loss)
I0726 17:36:50.343564  6123 solver.cpp:259]     Train net output #1: odm_loss = 0 (* 1 = 0 loss)
I0726 17:36:50.343580  6123 sgd_solver.cpp:138] Iteration 50, lr = 0.001
I0726 17:36:55.368263  6123 solver.cpp:243] Iteration 60, loss = nan
I0726 17:36:55.368419  6123 solver.cpp:259]     Train net output #0: arm_loss = nan (* 1 = nan loss)
I0726 17:36:55.368448  6123 solver.cpp:259]     Train net output #1: odm_loss = 0 (* 1 = 0 loss)

Why? is it the lmdb problem?

**Comment by Arthur-Shi** [comment]
I trained on VOC2007 datasets， is it possible normalization scale large？
Where can I modify？

**Comment by Arthur-Shi** [comment]
I tried： 
line323
normalizations = [5, 4, -1, -1]
and  line 243
base_lr = 0.0000004

It works, but
I0727 15:16:07.298811  4246 solver.cpp:243] Iteration 3420, loss = 14.3296
I0727 15:16:07.298864  4246 solver.cpp:259]     Train net output #0: arm_loss = 7.1132 (* 1 = 7.1132 loss)
I0727 15:16:07.298872  4246 solver.cpp:259]     Train net output #1: odm_loss = 10.5305 (* 1 = 10.5305 loss)
I0727 15:16:07.298879  4246 sgd_solver.cpp:138] Iteration 3420, lr = 1e-05
I0727 15:16:08.641118  4246 solver.cpp:243] Iteration 3430, loss = 9.724
I0727 15:16:08.641172  4246 solver.cpp:259]     Train net output #0: arm_loss = 0 (* 1 = 0 loss)
I0727 15:16:08.641182  4246 solver.cpp:259]     Train net output #1: odm_loss = 0 (* 1 = 0 loss)
I0727 15:16:08.641191  4246 sgd_solver.cpp:138] Iteration 3430, lr = 1e-05
I0727 15:16:10.040621  4246 solver.cpp:243] Iteration 3440, loss = 12.9493
I0727 15:16:10.040673  4246 solver.cpp:259]     Train net output #0: arm_loss = 7.29832 (* 1 = 7.29832 loss)
I0727 15:16:10.040683  4246 solver.cpp:259]     Train net output #1: odm_loss = 10.1251 (* 1 = 10.1251 loss)
I0727 15:16:10.040689  4246 sgd_solver.cpp:138] Iteration 3440, lr = 1e-05
I0727 15:16:11.454576  4246 solver.cpp:243] Iteration 3450, loss = 10.1832
I0727 15:16:11.454838  4246 solver.cpp:259]     Train net output #0: arm_loss = 0 (* 1 = 0 loss)
I0727 15:16:11.454860  4246 solver.cpp:259]     Train net output #1: odm_loss = 0 (* 1 = 0 loss)
I0727 15:16:11.454874  4246 sgd_solver.cpp:138] Iteration 3450, lr = 1e-05
I0727 15:16:12.832043  4246 solver.cpp:243] Iteration 3460, loss = 13.5415
I0727 15:16:12.832106  4246 solver.cpp:259]     Train net output #0: arm_loss = 4.10371 (* 1 = 4.10371 loss)
I0727 15:16:12.832120  4246 solver.cpp:259]     Train net output #1: odm_loss = 8.30004 (* 1 = 8.30004 loss)


Is it normal?
Is my modification correct? 
@sfzhang15

**Comment by sfzhang15** [comment]
@Arthur-Shi 
We have provided the training python file for VOC2007. Does you use it?

**Comment by Arthur-Shi** [comment]
ye, I am using python /RefineDet/examples/refinedet/VGG16_VOC2007_320.py.

Actually. I tried batch_size = 1, and base_lr = 0.0000004, normalizations = [5, 4, -1, -1].

This problem shown above appears fewer times, but still exist.

**Comment by sfzhang15** [comment]
@Arthur-Shi 
Why use batch_size = 1?

**Comment by Arthur-Shi** [comment]
@sfzhang15 my memory in my computer is not enough.
If I just have one gpu P4000 and memory 3.5G.
What is suitable for Batchsize？

**Comment by Arthur-Shi** [comment]
And also, did I modify below correctly?
normalizations = [5, 4, -1, -1]

**Comment by sfzhang15** [comment]
@Arthur-Shi 
How many images your 3.5G GPU can input?

**Comment by Arthur-Shi** [comment]
~$ free
             total       used       free     shared    buffers     cached
Mem:       8055988    5841156    2214832     120732     679916    2386032
-/+ buffers/cache:    2775208    5280780
Swap:      2000892     389200    1611692


~$ nvidia-smi
Tue Jul 31 16:39:26 2018       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 375.66                 Driver Version: 375.66                    |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  Quadro P4000        Off  | 0000:01:00.0      On |                  N/A |
| 52%   55C    P8    11W / 105W |    603MiB /  8112MiB |     26%      Default |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID  Type  Process name                               Usage      |
|=============================================================================|
|    0      1317    G   /usr/lib/xorg/Xorg                             286MiB |
|    0      2203    G   compiz                                         210MiB |
|    0     23291    G   ...el-token=58EB61A9F5FCF2C4EC8878F759B7794D   103MiB |
+-----------------------------------------------------------------------------+

I dont know how many images I can input.

**Comment by sfzhang15** [comment]
@Arthur-Shi 
You can try different batch_size to check how many images your GPU can input. Since you use only one GPU, so batch_size is the number of images input in your GPU.
